homeassistant/helpers/intent.py

```diff
@@ -66,6 +69,20 @@ def async_register(hass: HomeAssistant, handler: IntentHandler) -> None:
 
     intents[handler.intent_type] = handler
 
+    if handler.intent_type not in LLM_SKIP_AUTO_ADD_TOOL:
```

Comment:
Consider using `_slot_schema` or exposing it as public so this doesn't have to get into the specifics of specific intent handlers.

Reply:
I tried, but `_slot_schema` adds additional `{"value": validator}` layer that we don't want.
This is a weird thing about intents, they have a slot_schema of, for example,
```python
{
    vol.Any("name", "area", "floor"): cv.string,
    vol.Optional("domain"): vol.All(cv.ensure_list, [cv.string]),
    vol.Optional("device_class"): vol.All(cv.ensure_list, [cv.string]),
}
```
But what they actually accept is 
{
    vol.Any("name", "area", "floor"): {"value": cv.string},
    vol.Optional("domain"): {"value": vol.All(cv.ensure_list, [cv.string])},
    vol.Optional("device_class"): {"value": vol.All(cv.ensure_list, [cv.string])},
}
```
Sometimes even `{"value": cv.string, "text": Any}`.

Reply:
Ok, I can use `_slot_schema` and remove the extra layer in the `IntentTool`.

Reply:
Done